tracks/x86-64-assembly/exercises/two-fer/mentoring.md

```diff
@@ -0,0 +1,93 @@
+### Reasonable solutions
+
+Using a hand-rolled loop:
+
+```nasm
+    movzx edx, byte [rdi]
+    test dl, dl
+    je .L2
+.L1:
+    add rdi, 1
+    mov byte [rsi], dl
+    add rsi, 1
+    movzx edx, byte [rdi]
+    test dl, dl
+    jne .L1
+.L2:
+```
+
+Using the `rep movsb` instructions:
+
+```nasm
+    mov rdi, rsi
+    lea rsi, [rel symbol]
+    mov rcx, symbol_len
+    rep movsb
+```
+
+### Common errors
+
+- Forgetting to save registers with callee-save status (`rbx`, `rbp`, `r12-r15`).
```

Comment:
Perhaps we can show _why_ this is erroneous? What happens when you forget that? Does it still pass the tests? Why is it problematic?